### PR TITLE
planner, executor: adjustOverlongColName for CreateView

### PR DESCRIPTION
**Pull request by XuHuaiyu**
<!--
Thank you for contributing to TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
-->

### What problem does this PR solve? <!--add issue link with summary if exists-->
Fix the bug:
``` sql
On TiDB:
mysql root@127.0.0.1:test> create view v0 as select 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
(1059, "Identifier name 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' is too long")

On MySQL:
mysql root@127.0.0.1:test> create view v0 as select 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
Query OK, 0 rows affected
Time: 0.008s
mysql root@127.0.0.1:test> select * from v0
+-------------------------------------------------------------------+
| Name_exp_1 |
+-------------------------------------------------------------------+
| aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa |
+-------------------------------------------------------------------+
1 row in set
Time: 0.015s
```

### What is changed and how it works?

Rename the output column name for view if the column name is too long (>64). 


### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

 - Integration test


Code changes

 - Has exported function/method change

Side effects

- There are still some **compatible problems exist**. But I think the behavior of TiDB is more reasonable than MySQL, and there is no official MySQL manual taking about this behavior.

1. case 1:
In MySQL, use the sql shown in the result of `show create view` to create a view will fail. It is really strange. In TiDB, this case will fail for the first time.

``` sql
mysql> create view v(`aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`) as select a from t; -- TiDB will fail here.
Query OK, 0 rows affected (0.01 sec)

mysql> show create view v;
+------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| View | Create View                                                                                                                                                                                                                                                    | character_set_client | collation_connection |
+------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| v    | CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `v` AS select `t`.`a` AS `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` from `t` | utf8                 | utf8_general_ci      |
+------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
1 row in set (0.00 sec)

mysql> drop view v;
Query OK, 0 rows affected (0.00 sec)

mysql> CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `v` AS select `t`.`a` AS `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` from `t`;
ERROR 1166 (42000): Incorrect column name 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
```

case 2:

In MySQL, the following case will fail. But it will execute successfully in TiDB.
``` sql
create view v as select 'a' as 'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb' from t;
```

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Adjust the colName to name_exp_$off if a column name is overlong when creating the view.

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
@crazycs520 As I mentioned in the description. It's expected.

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -2948,3 +2947,15 @@ func buildChecksumTableSchema() (*expression.Schema, []*types.FieldName) {
...
 }
+
+// adjustOverlongViewColname adjusts the overlong outputNames of a view to
+// `new_exp_$off` where `$off` is the offset of the output column, $off starts from 1.
+// There is still some MySQL compatible problems.
+func adjustOverlongViewColname(plan LogicalPlan) {
+	outputNames := plan.OutputNames()
+	for i := range outputNames {
+		if outputName := outputNames[i].ColName.L; len(outputName) > mysql.MaxColumnNameLength {
+			outputNames[i].ColName = model.NewCIStr(fmt.Sprintf("name_exp_%d", i+1))
```
Comment (qw4990) [inline planner/core/planbuilder.go]: `name_exp_` should be capital?

Comment (XuHuaiyu) [reply planner/core/planbuilder.go]: I think it's ok to keep it lowercase here. It will not affect the execution of select.

**Pull request merged**